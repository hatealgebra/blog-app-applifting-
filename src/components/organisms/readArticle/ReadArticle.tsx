import React from "react";
import { components } from "../../../types/declarations";
import { ReadArticleContainer, ReadArticleContent } from "./readArticle.styled";

import ReactMarkdown from "react-markdown";

// TODO: Render image from the fetched image

const ReadArticle = ({
  title,
  imageSrc,
  author,
  createdAt,
  content,
}: components["schemas"]["ArticleDetail"] & {
  author: string;
  imageSrc: string;
}) => {
  const articlePublishedFormat = new Date(createdAt!)
    .toLocaleString()
    .split(" ")
    .slice(0, -1)
    .join(" ");
  return (
    <ReadArticleContainer className="read-article">
      <h1>{title}</h1>
      <div className="read-article__base-info label">
        <span className="read-article__author">{author}</span>
        <span>â€¢</span>
        <span className="read-article__date">{articlePublishedFormat}</span>
      </div>
      <img src={imageSrc} alt={`${title} image`} />
      <ReadArticleContent className="read-article__markdown">
        <ReactMarkdown children={content!} />
      </ReadArticleContent>
    </ReadArticleContainer>
  );
};

export default ReadArticle;
