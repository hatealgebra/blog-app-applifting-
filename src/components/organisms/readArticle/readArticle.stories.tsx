import React from 'react';

import ReadArticle from './ReadArticle';
import britishCat from '../../../images/british-haircat.jpg';

const content = `Men might grow a beard or maintain a mustache just for extra style points, but the facial hair of the cat has nothing to do with fashion. A cat's whiskers — or vibrissae — are a well-honed sensory tool that helps a catsee in the dark and steer clear of hungry predators. Whiskers
`;

export const ReadArticleExample = () => (
  <ReadArticle
    title="Why do cats have Whiskers?"
    imageSrc={britishCat}
    author="Elizabeth whatever"
    createdAt="12.04.2019"
    content={content}
  />
);

export default {
  title: 'Organisms/Read Article',
  component: ReadArticle,
};
