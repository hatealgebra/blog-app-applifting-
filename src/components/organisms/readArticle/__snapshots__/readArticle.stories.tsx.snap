// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Read Article ReadArticleExample smoke-test 1`] = `
<article class="sc-iAfmwU eGLYzN read-article">
  <h1>
    Why do cats have Whiskers?
  </h1>
  <div class="read-article__base-info label">
    <span class="read-article__author">
      Elizabeth whatever
    </span>
    <span>
      •
    </span>
    <span class="read-article__date">
      12/4/2019, 12:00:00
    </span>
  </div>
  <img src="static/media/src/images/british-haircat.jpg"
       alt="Why do cats have Whiskers? image"
  >
  <div class="sc-iUeHef read-article__markdown">
    <p>
      Men might grow a beard or maintain a mustache just for extra style points, but the facial hair of the cat has nothing to do with fashion. A cat's whiskers — or vibrissae — are a well-honed sensory tool that helps a catsee in the dark and steer clear of hungry predators.
    </p>
    <p>
      Whiskers are highly sensitive tactile hairs that grow in patterns on a cat's muzzle, above its eyes and elsewhere on its body, like the ears, jaw and forelegs. At the root of each of these long, stiff hairs is a follicle loaded with nerves.
    </p>
    <p>
      By brushing its whiskers against an object, a cat can detect the precise location, size and texture of the object, even in the dark. their surroundings, but can also provide humans with some insight into their pet's state of mind.
    </p>
    <p>
      For example, a set of taut whiskers, pulled back across the face, is a good indication that Kitty feels threatened, while relaxed whiskers, pointing away from the face, indicate a content cat. Of course, cats aren't the only mammals with whiskers.
    </p>
    <p>
      Most mammalian species, including primates, are equipped with these extrasensory receptors. Biologists think mammals developed whiskers because they needed help sensing their environments at night. The first small mammals shared the world with dinosaurs and had to adapt to hunting nocturnally, when their predators were less active.
    </p>
    <p>
      Whiskers helped these hungry animals find food and navigate dark terrain.
This evolutionary adaptation also helps to explain why the whiskers of many nocturnal or aquatic carnivores — like rats, seals and walruses — are so prominent.
    </p>
  </div>
</article>
`;
